src/spirc.rs

```diff
@@ -164,22 +182,21 @@ impl SpircManager {
             .state(state)
             .send();
     }
-
-    pub fn get_queue(&self) -> Vec<SpotifyId> {
-        self.0.lock().unwrap().tracks.clone()
-    }
 }
 
 impl SpircInternal {
     fn on_update(&mut self, player_state: &PlayerState) {
         let end_of_track = player_state.end_of_track();
         if end_of_track {
-            self.index = (self.index + 1) % self.tracks.len() as u32;
-            let track = self.tracks[self.index as usize];
-            self.player.load(track, true, 0);
-        } else {
-            self.notify_with_player_state(false, None, player_state);
+            if let Some(ref player) = self.player {
+                self.tracks.index = (self.tracks.index + 1) % self.tracks.ids.len() as u32;
+                let track = self.tracks.ids[self.tracks.index as usize];
+                player.load(track, true, 0);
+                return;
```

Comment:
Early returns like that are easy to miss.
Does it matter what happens in the no player case ?
Would that function ever be call anyway ?